mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,204 @@
+        #       NIRX also records "Study Type", "Experiment History",
+        #       "Additional Notes", "Contact Information"
+        subject_info = {}
+        subject_info['last_name'] = inf['name'].split()[-1].replace("\"", "")
+        subject_info['first_name'] = inf['name'].split()[0].replace("\"", "")
+        subject_info['middle_name'] = inf['name'].split()[-2].replace("\"", "")
+        subject_info['birthday'] = inf['age']
+        subject_info['sex'] = inf['gender'].replace("\"", "")
+        # Recode values
+        if subject_info['sex'] in {'M', 'Male', '1'}:
+            subject_info['sex'] = 1
+        if subject_info['sex'] in {'F', 'Female', '2'}:
+            subject_info['sex'] = 2
+        # NIRStar does not record an id, or handedness by default
+
+        # Read header file
+        # This is a bit tricky as the header file isnt compliant with
+        # the config specifications. So we need to remove all text
+        # between comments before passing to config parser
+        hdr_str = open(file_hdr[0]).read()
+        hdr_str = re.sub('#.*?#', '', hdr_str, flags=re.DOTALL)
+        hdr = cp.RawConfigParser()
+        hdr.read_string(hdr_str)
+
+        # Check that the file format version is supported
+        assert (hdr['GeneralInfo']['NIRStar'] == "\"15.2\""), \
+            "Only NIRStar version 15.2 is supported"
+
+        # Parse required header fields
+
+        # Extract source-detectors requested by user
+        sources = [int(s) for s in re.findall(r'(\d)-\d:\d+',
+                   hdr['DataStructure']['S-D-Key'])]
+        detectors = [int(s) for s in re.findall(r'\d-(\d):\d+',
+                     hdr['DataStructure']['S-D-Key'])]
+        sdindex = [int(s) for s in re.findall(r'\d-\d:(\d+)',
+                   hdr['DataStructure']['S-D-Key'])]
+        assert (len(sources) == len(detectors)), \
+            "Same amount of sources and detectors required"
+        assert (len(sources) == len(sdindex)), \
+            "Same amount of sources and keys required"
+
+        # Create mne structure
+        # TODO: ch_type is currently misc as I could not find appropriate
+        #       other type, the hbo and hbr type are not relevant until the
+        #       signal has been converted
+        # TODO: nchan needs to be multiplied by two as we have two wavelengths
+        #       per sensor should the underlying type be modified to
+        #       support (wavelength x channels x data)?
+        info = create_info(len(sources) * 2,
+                           hdr['ImagingParameters']['SamplingRate'],
+                           ch_types='misc')
+        info.update({'subject_info': subject_info})
+
+        # Store the subset of sources and detectors requested by user
+        # The signals between all source-detectors are stored even if they
+        # are meaningless, the user pre specifies which combinations are
+        # meaningful
+        # TODO: Is this style of info overloading allowed?
```

Comment:
No :) You are only allowed to use the entries in the `info` API page

https://mne.tools/dev/generated/mne.Info.html

The only output format we support is FIF, so anything you'd need for actual analysis we need to find a way to put in `Info` eventually.

There can be things that are specific to actually reading this data from disk, and those typically get stored in a separate dict that is file-format specific. This gets stored as `_raw_extras = my_extra_dict`. But it's temporary, in the sense that it's really just supposed to be for reading that format of file, it gets lost when you do `raw.save(fname); raw_read = mne.io.read_raw_fif(fname)`. So again, anything you'd need for actual analyses (like wavelengths!) would need to get stored in `Info` proper.

But for now since this is WIP it's fine, we can work on where to put these at some point. It would help if you could comment all these `fnirs_` entries with their ndarray shapes, so it's clear what's a singleton, what's 1 element per channel, multiple elements per channel, etc.

Reply:
That makes sense. Have removed additional info fields.